src/native/s3_client.c

```diff
@@ -67,7 +67,7 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientNew(
 
     struct aws_byte_cursor region = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_region);
 
-    struct s3_client_callback_data *callback_data = aws_mem_acquire(allocator, sizeof(struct s3_client_callback_data));
+    struct s3_client_callback_data *callback_data = aws_mem_calloc(allocator, sizeof(struct s3_client_callback_data), 1);
```

Comment:
params 2 and 3 are backwards

Reply:
A bit uneasy that this works. Will want to add tests to force some failures out - probably needing to make multiple requests in the same program execution.